addons/edi/models/edi_document.py

```diff
@@ -310,6 +357,42 @@ def _get_state_name(self):
         vals = dict(self.fields_get(allfields=["state"])["state"]["selection"])
         return vals[self.state]
 
+    @api.depends("waiting_time_start", "waiting_time_end")
+    def _compute_waiting_time(self):
+        """Compute the total time the document was waiting for processing"""
+        for rec in self:
+            if rec.waiting_time_start:
+                diff = (rec.waiting_time_end or fields.Datetime.now()) - rec.waiting_time_start
+                rec.waiting_time_total = diff.total_seconds() / 3600.0
```

Comment:
It might be better to have these total times in seconds rather than hours.

Reply:
Unfortunately, `float_time` widget requires the value in hours (see https://www.odoo.com/documentation/14.0/developer/reference/javascript/javascript_reference.html#non-relational-fields)